docs/proposals/crd-implementation.md

```diff
@@ -0,0 +1,50 @@
+# Service Catalog migration to CRDs
+
+The Service Catalog migration to CRDs was done in [this](https://github.com/kyma-project/kyma/issues/3034) epic. You can find there all of the tasks which were necessary to complete this challenge.
+
+## Implemented features
+
+The API server custom features were migrated to a admission webhook approach.
+
+The table converter API server feature was migrated to the [Additional Printer Columns](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/#additional-printer-columns) feature.
+
+## Project changes
+
+The architecture after migration looks as follows:
+
+<img src="/docs/images/current.png" width="75%" height="75%">
+
+### Chart changes
+
+The following files were added to the Service Catalog helm chart:
+
+- **crds.yaml** which specifies the CustomResourceDefinitions for the ServiceCatalog resources. The CustomResourceDefinitions are installed via helm hook:
+```
+"helm.sh/hook": crd-install
+```
+- **webhook-deployment.yaml** - webhook server deployment.
+- **webhook-register.yaml** - registration of the webhook server operations.
+- **webhook-service.yaml** - service for the webhook server.
+
```

Comment:
There will be extra file `cleaner-job.yaml` after merge PR #21 . It is responsible for create `pre-delete` job which makes sure all CRDs and CRs are removed after delete CR release.